src/main/java/com/microsoft/sqlserver/jdbc/SSPIAuthentication.java

```diff
@@ -149,23 +149,34 @@ String enrichSpnWithRealm(SQLServerConnection con, String spn, boolean allowHost
 
         String dnsName = m.group(1);
         String portOrInstance = m.group(2);
-        if (null == realm || realm.trim().isEmpty()) {
-            RealmValidator realmValidator = getRealmValidator();
-            realm = findRealmFromHostname(realmValidator, dnsName);
-            if (null == realm && allowHostnameCanonicalization) {
-                // We failed, try with canonical host name to find a better match
-                try {
+        try {
+            if (null == realm || realm.trim().isEmpty()) {
+                RealmValidator realmValidator = getRealmValidator();
+                realm = findRealmFromHostname(realmValidator, dnsName);
+                if (null == realm && allowHostnameCanonicalization) {
+                    if (logger.isLoggable(Level.FINER)) {
+                        logger.finer("Realm not provided or found from host name, attempt to find canonical host name via InetAddress function and then derive realm");
+                    }

```

Comment:
seems a bit wordy, how about "Attempt to derive realm using canonical host name"?